gazelle/generate.go

```diff
@@ -152,41 +147,86 @@ func (lang *JS) GenerateRules(args language.GenerateArgs) language.GenerateResul
+	generatedTestRules, generatedTestImports := lang.genJestTest(args, jsConfig, sources.jestSources)
 	generatedRules = append(generatedRules, generatedTestRules...)
 	generatedImports = append(generatedImports, generatedTestImports...)
 
-	appendTSExt := len(jsSources) > 0
+	appendTSExt := len(sources.jsSources) > 0
 
-	if len(jsSources) > 0 && jsConfig.CollectAll {
+	absJSRoot := path.Join(args.Config.RepoRoot, jsConfig.JSRoot)
+	isJSRoot := absJSRoot == args.Dir
+
+	if len(sources.jsSources) > 0 && jsConfig.CollectAll {
 		// add combined "ts_project" rule with js sources
-		generatedTSRules, generatedTSImports := lang.genRules(args, jsConfig, isBarrel, isJSRoot, pkgName, append(tsSources, jsSources...), false, "ts_project")
+		generatedTSRules, generatedTSImports := lang.genRules(
+			args,
+			jsConfig,
+			sources.isBarrel,
+			isJSRoot,
+			pkgName,
+			append(sources.tsSources, sources.jsSources...),
+			false,
+			"ts_project",
+		)
 		generatedRules = append(generatedRules, generatedTSRules...)
 		generatedImports = append(generatedImports, generatedTSImports...)
 
 	} else {
 		// add "ts_project" rule(s)
-		generatedTSRules, generatedTSImports := lang.genRules(args, jsConfig, isBarrel, isJSRoot, pkgName, tsSources, appendTSExt, "ts_project")
+		generatedTSRules, generatedTSImports := lang.genRules(
+			args,
+			jsConfig,
+			sources.isBarrel,
+			isJSRoot,
+			pkgName,
+			sources.tsSources,
+			appendTSExt,
+			"ts_project",
+		)
 		generatedRules = append(generatedRules, generatedTSRules...)
 		generatedImports = append(generatedImports, generatedTSImports...)
 
 		// add "js_library" rule(s)
 		appendTSExt = false
-		generatedJSRules, generatedJSImports := lang.genRules(args, jsConfig, isBarrel, isJSRoot, pkgName, jsSources, appendTSExt, "js_library")
+		generatedJSRules, generatedJSImports := lang.genRules(
+			args,
+			jsConfig,
+			sources.isBarrel,
+			isJSRoot,
+			pkgName,
+			sources.jsSources,
+			appendTSExt,
+			"js_library",
+		)
 		generatedRules = append(generatedRules, generatedJSRules...)
 		generatedImports = append(generatedImports, generatedJSImports...)
 	}
 
 	// add "web_assets" rule(s)
-	generatedWARules, generatedWAImports := lang.genWebAssets(args, webAssetsSet, jsConfig)
+	generatedWARules, generatedWAImports := lang.genWebAssets(
+		args,
+		sources.webAssetsSet,
+		jsConfig,
+	)
 	generatedRules = append(generatedRules, generatedWARules...)
 	generatedImports = append(generatedImports, generatedWAImports...)
 
 	// add "all_assets" "web_assets" rule
-	generatedAWARules, generatedAWAImports := lang.genAllAssets(args, isJSRoot, jsConfig)
+	generatedAWARules, generatedAWAImports := lang.genAllAssets(
+		args,
+		isJSRoot,
+		jsConfig,
+	)
 	generatedRules = append(generatedRules, generatedAWARules...)
 	generatedImports = append(generatedImports, generatedAWAImports...)
 
+	generatedTSRules, generatedTSImports := lang.genCollectedTargetsRule(
```

Comment:
I don't understand how referencing works, is calling this `generatedTSRules, generatedTSImports` make the imports to be overwritten